src/utils.py

```diff
@@ -67,9 +67,10 @@ def open_instagram(device_id):
     random_sleep()
 
 
-def close_instagram():
+def close_instagram(device_id):
```

Comment:
Are you sure that this will work?
We also have to call `close_instagram()` with this `device` argument everywhere.